### PR TITLE
[jvm-packages] fix "key not found: train" issue

### PR DESCRIPTION
**Pull request by wbo4958**

### CODE REVIEW

**Comment by wbo4958** [comment]
to fix https://github.com/dmlc/xgboost/issues/6806

**Comment by trivialfis** [comment]
@CodingCat Could you please help take a look?

File: jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala [hunk 1]
```diff
@@ -377,7 +377,7 @@ object XGBoost extends Serializable {
     // to workaround the empty partitions in training dataset,
     // this might not be the best efficient implementation, see
     // (https://github.com/dmlc/xgboost/issues/1277)
-    if (watches.toMap("train").rowNum == 0) {
```
Comment (CodingCat) [inline jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala:380]: is it possible that there is "train' as the key but it is empty?

I think we'd better keep both

Comment (wbo4958) [reply jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala:380]: toMap will filter out DMatrix with rowNum <= 0,  https://github.com/dmlc/xgboost/blob/master/jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala#L762

**Pull request merged**